### DISCUSSION TITLE
Feature Request: Add more animations.

### DISCUSSION THREAD

**Discussion by mathiasayivor** [vuesax-alpha/vuesax-alpha#57] [Ideas]
I stand to be corrected, but currently, there are only a few sets of animations. It would be nice to have more animation options available.

An example use for this would be the `Dialog` component. When it is in full-screen mode, using a slide transition works better than the current default animation (kinda like zoom-in)

**Comment by mathiasayivor** [comment]
Adding more animations means adding options to enable changing animations/transitions for each component that has one

E.g. via a `transition-show` and `transition-hide` prop.

**Comment by tranthinh-coding** [comment]
We can further animate state via `animate` props or via transition hooks.
Can you add this feature?

**Comment by mathiasayivor** [comment]
Unfortunately,  I am currently unavailable to implement this due to work. But I will sometime later.

**Comment by mathiasayivor** [comment]
Good job btw you did an awesome job in creating this project. I love the modern and unique feel.

**Comment by mathiasayivor** [comment]
@tranthinh-coding what do you think about adding animate.css?

**Comment by tranthinh-coding** [comment]
@mathiasayivor if it's transition animations like component show/hide actions,  in this project i added a few transition animations for `name` props of `<transition></transition>`.

**Comment by tranthinh-coding** [comment]
https://github.com/vuesax-alpha/vuesax-alpha/blob/eda8b28d0058411d3e7029037c629552b33fc9e1/packages/theme-chalk/src/common/transition.scss#L4

**Comment by tranthinh-coding** [comment]
@mathiasayivor As for animation effects, since each user's needs are different, I think to let users write their own `animated.css`.

**Comment by mathiasayivor** [comment]
Sure, but how easy is it for users to add their own animations?

**Comment by tranthinh-coding** [comment]
Of course. After adding animation props to the necessary components, the user can pass the name of that animation, ensure write the correct vue transition rules.

**Comment by mathiasayivor** [comment]
something like:

```vue
<vs-dialog v-model="active" animation="rotate-in">
<!-- dialog content -->
</vs-dialog>
```
?

**Comment by mathiasayivor** [comment]
Btw, I was just suggesting [animate.css](https://animate.style) as an easy way to add more animations since most users already know/use it.

**Comment by tranthinh-coding** [comment]
> cái gì đó như:
> 
> ```
> <vs-dialog v-model="active" animation="rotate-in">
> 
> </vs-dialog>
> ```
> 
> ?

That's right

**Comment by tranthinh-coding** [comment]
> Btw, I was just suggesting [animate.css](https://animate.style) as an easy way to add more animations since most users already know/use it.

Okay. Please help me do it. Create an animate.scss file [here](https://github.com/vuesax-alpha/vuesax-alpha/tree/dev/packages/theme-chalk/src/common), then use it [here](https://github.com/vuesax-alpha/vuesax-alpha/blob/eda8b28d0058411d3e7029037c629552b33fc9e1/packages/theme-chalk/src/base.scss#L3)

**Comment by tranthinh-coding** [comment]
Well, I think you can add more animate into transition.scss file.
Of course for advanced animations, use the animation property in the css. Let's write a new file separately
